File: app/src/main/java/mx/edu/upqroo/kristenandroid/data/database/daos/SubjectDao.java
package mx.edu.upqroo.kristenandroid.data.database.daos;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;
import mx.edu.upqroo.kristenandroid.data.database.entities.Subject;

@Dao
public interface SubjectDao {

    @Insert
    void insert(Subject subject);

    @Update
    void update(Subject... subjects);

    @Delete
    void delete(Subject... subjects);

    @Query("DELETE FROM subject")
    void deleteAll();

    @Query("SELECT * FROM subject WHERE subjectId = :subjectId")
    LiveData<Subject> getById(long subjectId);

    @Query("SELECT * FROM subject")
    LiveData<List<Subject>> getAll();

    @Query("SELECT * FROM subject WHERE dayId = :dayId")
    List<Subject> getByDayId(final long dayId);
}

File: app/src/main/java/mx/edu/upqroo/kristenandroid/data/database/daos/SubjectDao.kt
package mx.edu.upqroo.kristenandroid.data.database.daos

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import mx.edu.upqroo.kristenandroid.data.database.entities.Subject

@Dao
interface SubjectDao {

    @get:Query("SELECT * FROM subject")
    val all: LiveData<List<Subject>>

    @Insert
    fun insert(subject: Subject)

    @Update
    fun update(vararg subjects: Subject)

    @Delete
    fun delete(vararg subjects: Subject)

    @Query("DELETE FROM subject")
    fun deleteAll()

    @Query("SELECT * FROM subject WHERE subjectId = :subjectId")
    fun getById(subjectId: Long): LiveData<Subject>

    @Query("SELECT * FROM subject WHERE dayId = :dayId")
    fun getByDayId(dayId: Long): LiveData<List<Subject>>
}
